dp-cluster-setup-utility.py

```diff
@@ -1213,67 +1616,216 @@ def satisfied(self):
 
   def current_user(self):
     return pwd.getpwuid(os.getuid()).pw_name
+"""
+  Class for controlling flow of execution 
+"""
 
-if __name__ == '__main__':
-  user = Memorized(User())
-  print '\nThis script will check to ensure that all necessary pre-requisites have been met and then register this cluster with DataPlane.\n'
-  print 'Please ensure that your cluster has kerberos enabled, Ambari has been configured to use kerberos for authentication, and Knox is installed. Once those steps have been done, run this script from the Knox host and follow the steps and prompts to complete the cluster registration process.\n'
-
-  if not ScriptPrerequisites().satisfied():
-    sys.exit(1)
-
-  print 'Tell me about your DataPlane Instance'
-  dp = DataPlane(user.url_input('DataPlane URL', 'dp.url'), user.credential_input('DP Admin', 'dp.admin'))
+class FlowManager(object):
+  def __init__(self, cluster_type):
+    self.cluster_type = cluster_type
+    self.flow_manager = None
+    self.provider = None
+
+  def initialize_flow(self):
+    if self.cluster_type.name == 'ambari':
+      self.flow_manager = AmbariOperationsManager()
+      self.provider = 'AMBARI'
+    if self.cluster_type.name == 'cm':
+      self.flow_manager = CMOperationsManager()
+      self.provider = 'CM'
+  
+  def execute_flow(self):
+    cluster_instance = None
+    # Get cluster instance
+    try:
+      cluster_instance = self.flow_manager.get_instance()
+    except AttributeError as e:
+      raise e
+    
+    print('Working with cluster : %s' % cluster_instance.cluster.cluster_name)
+    
+    # Execute  pre_reqs check on the cluster
+    if not self.flow_manager.pre_reqs_satisfied(cluster_instance):
+      return 1
+    
+    print('Tell me about your DataPlane Instance')
+    dp = DataPlane(user.url_input('DataPlane URL', 'dp.url'), user.credential_input('DP Admin', 'dp.admin'),self.provider)
+    if dp.check_dependencies(cluster_instance.cluster, user):
+      print(dp.dependency_names())
+      return 1
+    
+    topology_util = self.flow_manager.get_topology_util(cluster_instance,dp)
```

Comment:
I think some of these methods don't belong to the current class. Let's take this one. The pseudo code for this method seems to be:
- For each dependency identified:
   - if cluster manager has this dependency:
      - map to a role name and collect

By passing cluster manager as a parameter, this whole code can be reused. No?